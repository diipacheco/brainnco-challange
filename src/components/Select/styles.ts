import styled from 'styled-components';

export const Container = styled.select`
  width: 100%;
  max-width: 215px;

  padding: 17px 33px;

  font-weight: 500;
  font-size: 15px;
  line-height: 18px;

  border-radius: 05px;

  background-color: var(--white);

  margin-top: 30px;
`;
