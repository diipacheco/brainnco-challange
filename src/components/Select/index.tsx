import React, { useEffect } from 'react';

import { useLotteries } from '../../hooks/lotteries';

import { Container } from './styles';

const Select: React.FC = () => {
  const { lotteries, fetchLotteries, handleSelectedLottery } = useLotteries();

  useEffect(() => {
    fetchLotteries();
  }, [fetchLotteries]);

  return (
    <>
      <Container
        data-testid="select"
        onChange={event => handleSelectedLottery(event.target.value)}
      >
        {lotteries?.map(lottery => (
          <option
            data-testid={`selected-option-${lottery.id}`}
            key={lottery.id}
            value={lottery.nome}
          >
            {lottery.nome?.toUpperCase()}
          </option>
        ))}
      </Container>
    </>
  );
};

export default Select;
