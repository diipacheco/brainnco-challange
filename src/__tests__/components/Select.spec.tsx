import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import Select from '../../components/Select';

jest.mock('../../hooks/lotteries', () => ({
  useLotteries: () => ({
    lotteries: [
      {
        id: 0,
        nome: 'mega-sena',
      },
      {
        id: 1,
        nome: 'quina',
      },
      {
        id: 2,
        nome: 'lotofÃ¡cil',
      },
      {
        id: 3,
        nome: 'lotomania',
      },
      {
        id: 4,
        nome: 'timemania',
      },
      {
        id: 5,
        nome: 'dia de sorte',
      },
    ],
    selectedLottery: 'MEGA-SENA',
    fetchLotteries: jest.fn(),
    handleSelectedLottery: jest.fn(),
  }),
}));

describe('Select', () => {
  it('should match snapshots', () => {
    const wrapper = render(<Select />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should be able to render all lotteries options', () => {
    const { getByTestId } = render(<Select />);

    const selectElement = getByTestId('select');

    expect(selectElement).toHaveLength(6);
  });

  it('should be able to select a specific lottery', () => {
    const { getByTestId } = render(<Select />);

    const selectElement = getByTestId('select');

    userEvent.selectOptions(selectElement, 'QUINA');

    expect(
      (getByTestId('selected-option-0') as HTMLOptionElement).selected,
    ).toBeFalsy();
    expect(
      (getByTestId('selected-option-1') as HTMLOptionElement).selected,
    ).toBeTruthy();
  });
});
