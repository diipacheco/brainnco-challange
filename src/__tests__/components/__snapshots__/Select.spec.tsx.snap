// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select should match snapshots 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  width: 100%;
  max-width: 215px;
  padding: 17px 33px;
  font-weight: 500;
  font-size: 15px;
  line-height: 18px;
  border-radius: 05px;
  background-color: var(--white);
  margin-top: 30px;
}

<body>
    <div>
      <select
        class="c0"
        data-testid="select"
      >
        <option
          data-testid="selected-option-0"
          value="mega-sena"
        >
          MEGA-SENA
        </option>
        <option
          data-testid="selected-option-1"
          value="quina"
        >
          QUINA
        </option>
        <option
          data-testid="selected-option-2"
          value="lotofácil"
        >
          LOTOFÁCIL
        </option>
        <option
          data-testid="selected-option-3"
          value="lotomania"
        >
          LOTOMANIA
        </option>
        <option
          data-testid="selected-option-4"
          value="timemania"
        >
          TIMEMANIA
        </option>
        <option
          data-testid="selected-option-5"
          value="dia de sorte"
        >
          DIA DE SORTE
        </option>
      </select>
    </div>
  </body>,
  "container": <div>
    <select
      class="sc-bdnxRM heiljH"
      data-testid="select"
    >
      <option
        data-testid="selected-option-0"
        value="mega-sena"
      >
        MEGA-SENA
      </option>
      <option
        data-testid="selected-option-1"
        value="quina"
      >
        QUINA
      </option>
      <option
        data-testid="selected-option-2"
        value="lotofácil"
      >
        LOTOFÁCIL
      </option>
      <option
        data-testid="selected-option-3"
        value="lotomania"
      >
        LOTOMANIA
      </option>
      <option
        data-testid="selected-option-4"
        value="timemania"
      >
        TIMEMANIA
      </option>
      <option
        data-testid="selected-option-5"
        value="dia de sorte"
      >
        DIA DE SORTE
      </option>
    </select>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
